src/Altinn.Platform/Altinn.Platform.Register/Register/Services/Implementation/PartiesWrapper.cs

```diff
@@ -21,30 +21,41 @@ public class PartiesWrapper : IParties
         private readonly GeneralSettings _generalSettings;
         private readonly ILogger _logger;
         private readonly HttpClient _client;
+        private readonly IMemoryCache _parties;
+        private const int _cacheTimeout = 5;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="PartiesWrapper"/> class
         /// </summary>
         /// <param name="httpClient">HttpClient from default httpclientfactory</param>
         /// <param name="generalSettings">the general settings</param>
         /// <param name="logger">the logger</param>
-        public PartiesWrapper(HttpClient httpClient, IOptions<GeneralSettings> generalSettings, ILogger<PartiesWrapper> logger)
+        /// <param name="partiesCache">memory cache for parties</param>
+        public PartiesWrapper(HttpClient httpClient, IOptions<GeneralSettings> generalSettings, ILogger<PartiesWrapper> logger, IMemoryCache partiesCache)
```

Comment:
Maybe rename the MemoryCachee to something a little bit more generic. The MemoryCache service is a singleton. If other classes starts using it for caching it might hold a lot more than parties.